salt/preflight/mandatory.sls

```diff
@@ -0,0 +1,63 @@
+{%- from 'preflight/map.jinja' import settings with context %}
+
+install mandatory packages:
+  pkg.installed:
+    - pkgs: {{ settings.mandatory.packages }}
+
+{%- if settings.destructive %}
+{%- for directory in settings.mandatory.available_pathdirs %}
+make sure {{ directory }} does not exist:
+  file.absent:
+    - name: {{ directory }}
+{%- endfor %}
+{%- endif %}
+
+{%- for service in settings.mandatory.services.running %}
+enable {{ service  }} service:
+  service.running:
+    - name: {{ service  }}
+    - enable: true
+{%- endfor %}
+
+# required to set sysctl net.bridge.bridge-nf-call-iptables
+add the module br_netfilter to kernel:
+  kmod.present:
+    - name: br_netfilter
+    - persist: True
+
+{%- for item, value in settings.mandatory.sysctl_values.items() %}
+set sysctl {{ item }} value to {{ value }}:
+  sysctl.present:
+    - name: {{ item }}
+    - value: {{ value }}
+    - config: /etc/sysctl.d/60-metalk8s.conf
+    {%- if item in ("net.bridge.bridge-nf-call-ip6tables", "net.bridge.bridge-nf-call-iptables") %}
+    - require:
+      - kmod: add the module br_netfilter to kernel
+    {%- endif %}
+{%- endfor %}
+
+disable devices and files for paging and swapping:
+  cmd.run:
+    - name: swapoff -a
+    - onlyif:
+      - swapon --show=NAME | grep -q NAME
+
+prevent swap mount from fstab:
+  file.comment:
+    - name: /etc/fstab
+    - regex: .*\s+swap\s+.*
```

Comment:
That would comment all lines that happen to have "swap" in their name.
You need to test only the third column for the value

Reply:
Maybe [salt.modules.mount](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.mount.html) can help (`swaps` to list & `swapoff` to disable).

Reply:
for swapoff yes, (see my comment above).
Sadly, there is nothing to "ensure" that a swap device is unmounted.
I tested and for example, mount.swaps retrun the device mapper as key, while yu have for example the LVM path in /etc/fstab, so you can't rely on it.
```
$ sudo salt-call --local mount.swaps
local:                              
    ----------                      
    /dev/dm-3:                      
        ----------                  
        priority:                   
            -2                      
        size:                       
            16428540                
        type:                       
            partition               
        used:                       
            0                       
$ grep swap /etc/fstab                                                          
/dev/mapper/cryptswap1 none swap sw 0 0                                         
```

Reply:
O, great, my whole example with mount.swaps and swapoff didn't make it when I submitted the review ...